<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.6.42">

  <meta name="author" content=" SURE 2024 Department of Statistics &amp; Data Science Carnegie Mellon University">
  <title>SURE 2024 – Unsupervised learning: Gaussian mixture models</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-62cb22a9f2c402427105faaaa1e9cd8b.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Unsupervised learning: Gaussian mixture models</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
<br>SURE 2024<br><br>Department of Statistics &amp; Data Science<br>Carnegie Mellon University 
</div>
</div>
</div>

</section>
<section>
<section id="background" class="title-slide slide level1 center">
<h1>Background</h1>

</section>
<section id="previously-clustering" class="slide level2">
<h2>Previously: clustering</h2>
<ul>
<li><p>Goal: partition <span class="math inline">\(n\)</span> data points into <span class="math inline">\(K\)</span> subgroups, where roughly points within group are more “similar” than points between groups</p></li>
<li><p>Previous methods: <strong><span class="math inline">\(k\)</span>-means clustering </strong> and <strong>hierarchical clustering</strong></p></li>
</ul>
<div class="fragment">
<ul>
<li><strong>Classification vs clustering</strong></li>
</ul>
<p><strong>Classification</strong></p>
<ul>
<li><p>We are given labeled (categorical) data</p></li>
<li><p>Focus on generalization (i.e.&nbsp;we want to have a classifier that performs well on predicting new (test) data)</p></li>
</ul>
</div>
<div class="fragment">
<p><strong>Clustering</strong></p>
<ul>
<li><p>We are only given the points (with no labels)</p></li>
<li><p>We want to find interesting subgroups/structure in the data</p></li>
</ul>
</div>
</section>
<section id="previously-clustering-1" class="slide level2">
<h2>Previously: clustering</h2>
<ul>
<li><p>Previous methods: <strong><span class="math inline">\(k\)</span>-means clustering </strong> and <strong>hierarchical clustering</strong></p></li>
<li><p>Output <strong>hard</strong> assignments, strictly assigning observations to only one cluster</p></li>
</ul>
<div class="fragment">
<ul>
<li><p>What about <strong>soft</strong> assignments and <strong>uncertainty</strong> in the clustering results?</p>
<ul>
<li><p>Assigns each observation a probability of belonging to a cluster</p></li>
<li><p>Incorporate statistical modeling into clustering</p></li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>We want to estimate the density of the observations in some way that allows us to extract clusters</li>
</ul>
</div>
</section>
<section id="motivating-figure" class="slide level2">
<h2>Motivating figure</h2>
<p><br></p>

<img data-src="https://miro.medium.com/v2/resize:fit:1100/format:webp/0*uQTamSp8hAcnJPl0." class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="model-based-clustering" class="slide level2">
<h2>Model-based clustering</h2>
<p>Key idea: data are considered as coming from a mixture of underlying probability distributions</p>
<div class="fragment">
<p>Most popular method: Gaussian mixture model (GMM)</p>
<ul>
<li><p>Each observation is assumed to be distributed as one of <span class="math inline">\(k\)</span> multivariate normal distributions</p></li>
<li><p><span class="math inline">\(k\)</span> is the number of clusters (components)</p></li>
</ul>
</div>
</section>
<section id="previously-kernel-density-estimation" class="slide level2">
<h2>Previously: kernel density estimation</h2>
<p>Kernel density estimator: <span class="math inline">\(\displaystyle \hat{f}(x) = \frac{1}{n} \sum_{i=1}^n \frac{1}{h} K_h(x - x_i)\)</span></p>
<div class="fragment">
<div class="columns">
<div class="column" style="text-align: left;">
<ul>
<li><p>Smooth each data point into a small density bumps</p></li>
<li><p>Sum all these small bumps together to obtain the final density estimate</p></li>
</ul>
<p>Use <strong>every observation</strong> when estimating the density for new points</p>
<p>We want to estimate the density of the points in some way that allows us to extract clusters</p>
</div><div class="column" style="text-align: left;">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../images/kde-bumps.png" class="quarto-figure quarto-figure-center" width="678"></p>
</figure>
</div>
</div>
</div>
</div></div>
</div>
</section>
<section id="mixture-models" class="slide level2">
<h2>Mixture models</h2>
<ul>
<li>Model the density as a mixture of distributions</li>
</ul>

<img data-src="https://angusturner.github.io/assets/images/mixture.png" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="mixture-models-1" class="slide level2">
<h2>Mixture models</h2>
<p>Formally, <span class="math display">\[\displaystyle f(x) = \sum_{k=1}^K \tau_k f_k(x)\]</span></p>
<p>where</p>
<ul>
<li><p><span class="math inline">\(f_k\)</span> are some distributions</p></li>
<li><p><span class="math inline">\(\tau_k \ge 0\)</span> and <span class="math inline">\(\sum_{k=1}^K \tau_k = 1\)</span> are the mixture weights</p></li>
<li><p><span class="math inline">\(K\)</span> is the number of mixture components (i.e.&nbsp;the number of clusters)</p></li>
</ul>
</section>
<section id="mixture-models-2" class="slide level2">
<h2>Mixture models</h2>
<p>This is a <strong>data generating process</strong></p>
<p>Imagine that each cluster (component) has a different distribution</p>
<div class="fragment">
<p>Task: Generate a new observation (i.e.&nbsp;sample) from a mixture model</p>
</div>
<div class="fragment">
<ul>
<li><p>Choose a cluster by drawing <span class="math inline">\(Z \in  \{1, \dots, K\}\)</span>, where <span class="math inline">\(P(Z = k) = \tau_k\)</span></p>
<ul>
<li>i.e.&nbsp;<span class="math inline">\(Z\)</span> is a categorical latent variable indicating which cluster the new observation is from</li>
</ul></li>
<li><p>Generate an observation from the distribution <span class="math inline">\(f_Z\)</span> (corresponding to cluster <span class="math inline">\(Z\)</span>)</p></li>
</ul>
</div>
<div class="fragment">
<p>Hence <span class="math inline">\(\displaystyle f(x) = \sum_{k=1}^K P(Z = k) p (x \mid Z = k) = \sum_{k=1}^K \tau_k f_Z(x)\)</span></p>
<!-- . . . -->
<!-- 1. __Pick a distribution/component__ among our $K$ options by introducing a new variable $$z \sim \text{Multinomial} (\tau_1, \tau_2, \dots, \tau_k)$$ i.e. $z$ is a categorical latent variable indicating which component the new point is from -->
<!-- . . . -->
<!-- 2. __Generate an observation with that distribution/component__, i.e. $x \mid z \sim f_{z}$ -->
<!-- . . . -->
<!-- _So what do we use for each $f_k$?_ -->
</div>
</section>
<section id="gaussian-mixture-models-gmms" class="slide level2">
<h2>Gaussian mixture models (GMMs)</h2>
<ul>
<li><p>Assume a <strong>parametric mixture model</strong>, with <strong>parameters</strong> <span class="math inline">\(\theta_k\)</span> for the <span class="math inline">\(k\)</span>th component <span class="math display">\[f(x) = \sum_{k=1}^K \tau_k f_k(x; \theta_k)\]</span></p></li>
<li><p>Gaussian mixture models (GMMs) are perhaps the most popular mixture models</p></li>
</ul>
<div class="fragment">
<ul>
<li>Assume each component is <a href="https://en.wikipedia.org/wiki/Normal_distribution">Gaussian (normal)</a>, where the 1D case is</li>
</ul>
<p><span class="math display">\[f_k(x; \theta_k) = N(x; \mu_k, \sigma_k^2) = \frac{1}{\sqrt{2 \pi \sigma_k^2}}\text{exp} \left( -\frac{(x - \mu_k)^2}{2 \sigma_k^2} \right)\]</span></p>
<p>with mean <span class="math inline">\(\mu_k\)</span> and variance <span class="math inline">\(\sigma_k ^2\)</span></p>
</div>
<div class="fragment">
<ul>
<li>We need to estimate each <span class="math inline">\(\{\tau_1, \dots, \tau_k\}\)</span>, <span class="math inline">\(\{\mu_1, \dots, \mu_k\}\)</span>, <span class="math inline">\(\{\sigma_1, \dots, \sigma_k\}\)</span></li>
</ul>
</div>
</section>
<section id="gaussian-mixture-models-gmms-1" class="slide level2">
<h2>Gaussian mixture models (GMMs)</h2>

<img data-src="https://miro.medium.com/v2/resize:fit:1100/format:webp/0*uQTamSp8hAcnJPl0." class="quarto-figure quarto-figure-center r-stretch" style="width:60.0%"><p>Model <span class="math inline">\(\tau_1, \tau_2, \tau_3\)</span> and</p>
<p><span class="math inline">\(f_{\text{red}}(x) = N(\mu_{\text{red}}, \sigma^2_{\text{red}})\)</span></p>
<p><span class="math inline">\(f_{\text{blue}}(x) = N(\mu_{\text{blue}}, \sigma^2_{\text{blue}})\)</span></p>
<p><span class="math inline">\(f_{\text{green}}(x) = N(\mu_{\text{green}}, \sigma^2_{\text{green}})\)</span></p>
</section>
<section id="estimating-a-mixture-model" class="slide level2">
<h2>Estimating a mixture model</h2>
<p>What can we do to soft-cluster our data?</p>
<div class="fragment">
<p>For an observation <span class="math inline">\(x\)</span>, compute the weight for <span class="math inline">\(x\)</span> to belong to cluster <span class="math inline">\(k\)</span> for <span class="math inline">\(k \in \{1, \dots, K\}\)</span> <span class="math display">\[ P(Z = k \mid X = x) = \frac{P(X = x \mid Z = k) P(Z = k)}{\sum_{j}P(X = x \mid Z = j) P(Z = j)} =\frac{\tau_{k} \ N(\mu_{k}, \sigma_{k}^{2})}{\sum_{j=1}^{K} \tau_{j} \ N(\mu_{j}, \sigma^2_{j})}\]</span></p>
<p>(recall that <span class="math inline">\(\displaystyle f(x) = \sum_{k=1}^K \tau_k \ N(\mu_k, \sigma^2_k)\)</span>)</p>
</div>
<div class="fragment">
<p>How do we estimate the parameters of the mixture model (i.e.&nbsp;the <span class="math inline">\(\tau_k, \mu_k, \sigma_k\)</span>)?</p>
</div>
<div class="fragment">
<ul>
<li><p>We don’t know which component (i.e.&nbsp;the true labels) an observation actually belongs to</p></li>
<li><p><span class="math inline">\(Z\)</span> is a latent variable (since the true cluster labels are not observed)</p></li>
<li><p>This is known as the missing data or latent variable problems</p></li>
</ul>
</div>
</section>
<section id="estimating-a-mixture-model-maximum-likelihood" class="slide level2">
<h2>Estimating a mixture model: maximum likelihood</h2>
<p>Likelihood function: how likely to observe data for given parameter values</p>
<div class="fragment">
<p>Setup: Given observations <span class="math inline">\(X_1, \dots, X_n\)</span></p>
<p>Task: Estimate the parameters of a mixture model</p>
</div>
<div class="fragment">
<ul>
<li>What if we know the true cluster labels <span class="math inline">\(Z_1, \dots, Z_n\)</span>? We can just use maximum likelihood estimation and maximize <span class="math display">\[L\{\left(\tau_k, \mu_k, \sigma^2_k \right)_{k=1}^K\} = \sum_{i=1}^n \log f(X_i, Z_i)\]</span> (just take data from each group, compute the fraction of points <span class="math inline">\(\tau_k\)</span>, the mean <span class="math inline">\(\mu_k\)</span>, and the variance <span class="math inline">\(\sigma^2\)</span>)</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>In our case, since we do not know the cluster labels <span class="math inline">\(Z_i\)</span>, we should try to maximize the (marginal) likelihood of the observed data</li>
</ul>
<p><span class="math display">\[\mathcal L\{\left(\tau_k, \mu_k, \sigma^2_k \right)_{k=1}^K\} = \sum_{i=1}^n \log f(X_i) = \sum_{i=1}^n \log \left[\sum_{k=1}^K f(X_i, k)\right]\]</span></p>
</div>
</section>
<section id="estimating-a-mixture-model-em-algorithm" class="slide level2">
<h2>Estimating a mixture model: EM algorithm</h2>
<p><a href="https://en.wikipedia.org/wiki/Expectation%E2%80%93maximization_algorithm">Expectation–maximization (EM) algorithm</a> is a method for (approximately) maximizing the (marginal) likelihood in the presence of missing data</p>
<p>For a GMM:</p>
<ul>
<li><p>First, initialize the model parameters randomly</p></li>
<li><p>Then, alternate between the following two steps (keep repeating until nothing changes)</p>
<ul>
<li><p><strong>E-step</strong>: compute the cluster memberships for each point</p></li>
<li><p><strong>M-step</strong>: recompute/update the parameters</p></li>
</ul></li>
</ul>
<!-- ## Let's pretend we only have one component... -->
<!-- Set up: Given $n$ observations from a single normal distribution -->
<!-- Task: Estimate the distribution parameters using the __likelihood function__ - the probability/density of observing the data given the parameters $$\mathcal{L}(\mu, \sigma \mid x_1, \dots, x_n) = f( x_1, \dots, x_n \mid \mu, \sigma) =  \prod_{i=1}^n \frac{1}{\sqrt{2\tau \sigma^2}}\text{exp } \left\{-\frac{(x_i - \mu)^2}{2 \sigma^2}\right\}$$ -->
<!-- . . . -->
<!-- Compute the __maximum likelihood estimates (MLEs)__ for $\mu$ and $\sigma$ -->
<!-- - $\displaystyle \hat{\mu}_{\rm MLE} = \frac{1}{n} \sum_i^n x_i$, sample mean -->
<!-- - $\displaystyle \hat{\sigma}_{\rm MLE} = \sqrt{\frac{1}{n}\sum_i^n (x_i - \mu)^2}$, sample standard deviation (plug in $\hat{\mu}_{\rm MLE}$) -->
<!-- ## The problem with more than one component -->
<!-- ::: columns -->
<!-- ::: {.column width="50%" style="text-align: left;"} -->
<!-- - __We don't know which component an observation belongs to__ -->
<!-- - __IF WE DID KNOW__, then we could compute each component's MLEs as before -->
<!-- - But we don't know because $z$ is a __latent variable__! So what about its distribution given the data? -->
<!-- $\displaystyle P(z_i = k \mid x_i) = \frac{P(x_i \mid z_i = k) P(z_i = k)}{P(x_i)}$ -->
<!-- $\displaystyle =\frac{\tau_{k} N\left(\mu_{k}, \sigma_{k}^{2}\right)}{\sum_{k=1}^{K} \tau_{k} N\left(\mu_{k}, \sigma_{k}\right)}$ -->
<!-- - __But we do NOT know these parameters!__ -->
<!-- ::: -->
<!-- ::: {.column width="50%" style="text-align: left;"} -->
<!-- ```{r init-sim-data, echo = FALSE, fig.align='center', fig.height=9} -->
<!-- library(tidyverse) -->
<!-- # mixture components -->
<!-- mu_true <- c(5, 13) -->
<!-- sigma_true <- c(1.5, 2) -->
<!-- # determine Z_i -->
<!-- z <- rbinom(500, 1, 0.75) -->
<!-- # sample from mixture model -->
<!-- x <- rnorm(10000, mean = mu_true[z + 1],  -->
<!--            sd = sigma_true[z + 1]) -->
<!-- tibble(xvar = x) |> -->
<!--   ggplot(aes(x = xvar)) + -->
<!--   geom_histogram(color = "black", -->
<!--                  fill = "darkblue", -->
<!--                  alpha = 0.3) + -->
<!--   labs(x = "Simulated variable", -->
<!--        y = "Count") -->
<!-- ``` -->
<!-- ::: -->
<!-- ::: -->
</section>
<section id="expectation-maximization-em-algorithm" class="slide level2">
<h2>Expectation-maximization (EM) algorithm</h2>
<p>Alternate between the following:</p>
<ul>
<li><em>pretend</em> to know the probability each observation belongs to each group, to estimate the parameters of the components</li>
</ul>
<div class="fragment">
<ul>
<li><em>pretend</em> to know the parameters of the components, to estimate the probability each observation belong to each group</li>
</ul>
</div>
<div class="fragment">
<ol type="1">
<li><p>Start with initial guesses about <span class="math inline">\(\tau_1, \dots, \tau_k\)</span>, <span class="math inline">\(\mu_1, \dots, \mu_k\)</span>, <span class="math inline">\(\sigma_1, \dots, \sigma_k\)</span></p></li>
<li><p>Repeat until nothing changes:</p></li>
</ol>
</div>
<div class="fragment">
<ul>
<li><p><strong>E-step</strong>: calculate <span class="math inline">\(\hat{z}_{ik}\)</span>, the weight for observation <span class="math inline">\(i\)</span> belonging to cluster <span class="math inline">\(k\)</span></p></li>
<li><p><strong>M-step</strong>: update parameter estimates with <strong>weighted</strong> MLE using <span class="math inline">\(\hat{z}_{ik}\)</span></p></li>
</ul>
</div>
</section>
<section id="is-this-familiar" class="slide level2">
<h2>Is this familiar?</h2>
<div class="fragment">
<ul>
<li>In GMMs, we’re essentially guessing the latent variables <span class="math inline">\(Z_i\)</span> and pretending to know the parameters to perform maximum likelihood estimation</li>
</ul>
</div>
<div class="fragment">
<div class="columns">
<div class="column" style="text-align: left;">
<ul>
<li><p>This resembles the <span class="math inline">\(k\)</span>-means algorithm!</p>
<ul>
<li><p>The cluster centroids are chosen at random, and then recomputed/updated</p></li>
<li><p>This is repeated until the cluster assignments stop changing</p></li>
</ul></li>
</ul>
</div><div class="column" style="text-align: left;">
<center>
<iframe src="https://giphy.com/embed/kd9BlRovbPOykLBMqX" width="270" height="270" frameborder="0" class="giphy-embed" allowfullscreen="">
</iframe>
</center>
</div></div>
</div>
<div class="fragment">
<ul>
<li>Instead of assigning each point to a single cluster, we “softly” assign them so they contribute fractionally to each cluster</li>
</ul>
</div>
</section>
<section id="how-does-this-relate-back-to-clustering" class="slide level2">
<h2>How does this relate back to clustering?</h2>
<div class="fragment">
<p>From the EM algorithm: <span class="math inline">\(\hat{z}_{ik}\)</span> is the estimated weight for observation <span class="math inline">\(i\)</span> belonging to cluster <span class="math inline">\(k\)</span> (i.e.&nbsp;soft membership)</p>
</div>
<div class="fragment">
<ul>
<li><p>assign observation <span class="math inline">\(i\)</span> to a cluster with the largest <span class="math inline">\(\hat{z}_{ik}\)</span></p></li>
<li><p>measure cluster assignment <strong>uncertainty</strong> of <span class="math inline">\(\displaystyle 1 - \max_k \hat{z}_{ik}\)</span></p></li>
</ul>
</div>
<div class="fragment">
<p><strong>Our parameters determine the type of clusters</strong></p>
</div>
<div class="fragment">
<p>In the 1D case, there are two options:</p>
</div>
<div class="fragment">
<ol type="1">
<li>each cluster <strong>is assumed to have equal variance</strong> (spread): <span class="math inline">\(\sigma_1^2 = \sigma_2^2 = \dots = \sigma_k^2\)</span></li>
</ol>
</div>
<div class="fragment">
<ol start="2" type="1">
<li>each cluster <strong>is allowed to have a different variance</strong></li>
</ol>
</div>
<div class="fragment">
<p><em>But that is only 1D… what happens in multiple dimensions?</em></p>
</div>
</section>
<section id="multivariate-gmms" class="slide level2">
<h2>Multivariate GMMs</h2>
<p><span class="math display">\[f(x) = \sum_{k=1}^K \tau_k f_k(x; \theta_k) \qquad \text{where }f_k(x; \theta_k) \sim N(\boldsymbol{\mu}_k, \boldsymbol{\Sigma}_k)\]</span></p>
<p>Each component is a <strong>multivariate normal distribution</strong>:</p>
<div class="fragment">
<ul>
<li><span class="math inline">\(\boldsymbol{\mu}_k\)</span> is a <em>vector</em> of means in <span class="math inline">\(p\)</span> dimensions</li>
</ul>
</div>
<div class="fragment">
<ul>
<li><span class="math inline">\(\boldsymbol{\Sigma}_k\)</span> is the <span class="math inline">\(p \times p\)</span> <strong>covariance</strong> matrix - describes the joint variability between pairs of variables</li>
</ul>
<p><span class="math display">\[\sum=\left[\begin{array}{cccc}
\sigma_{1}^{2} &amp; \sigma_{1,2} &amp; \cdots &amp; \sigma_{1, p} \\
\sigma_{2,1} &amp; \sigma_{2}^{2} &amp; \cdots &amp; \sigma_{2, p} \\
\vdots &amp; \vdots &amp; \ddots &amp; \vdots \\
\sigma_{p, 1} &amp; \sigma_{p, 2}^{2} &amp; \cdots &amp; \sigma_{p}^{2}
\end{array}\right]\]</span></p>
</div>
</section>
<section id="covariance-constraints" class="slide level2">
<h2>Covariance constraints</h2>
<p><span class="math display">\[\sum=\left[\begin{array}{cccc}
\sigma_{1}^{2} &amp; \sigma_{1,2} &amp; \cdots &amp; \sigma_{1, p} \\
\sigma_{2,1} &amp; \sigma_{2}^{2} &amp; \cdots &amp; \sigma_{2, p} \\
\vdots &amp; \vdots &amp; \ddots &amp; \vdots \\
\sigma_{p, 1} &amp; \sigma_{p, 2}^{2} &amp; \cdots &amp; \sigma_{p}^{2}
\end{array}\right]\]</span></p>
<div class="fragment">
<p>As the number of dimensions increases, model fitting and estimation become increasingly difficult</p>
</div>
<div class="fragment">
<p>We can use <strong>constraints</strong> on multiple aspects of the <span class="math inline">\(k\)</span> covariance matrices:</p>
</div>
<div class="fragment">
<ul>
<li><p><strong>volume</strong>: size of the clusters, i.e., number of observations,</p></li>
<li><p><strong>shape</strong>: direction of variance, i.e.&nbsp;which variables display more variance</p></li>
<li><p><strong>orientation</strong>: aligned with axes (low covariance) versus tilted (due to relationships between variables)</p></li>
</ul>
</div>
</section>
<section id="covariance-constraints-1" class="slide level2">
<h2>Covariance constraints</h2>

<img data-src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5096736/bin/nihms793803f2.jpg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"><ul>
<li><p>The three letters in the model name denote, in order, the volume, shape, and orientation across clusters</p></li>
<li><p><strong>E</strong>: equal and <strong>V</strong>: varying (<strong>VVV</strong> is the most flexible, but has the most parameters)</p></li>
<li><p>Two II: <strong>spherical</strong>, one I: <strong>diagonal</strong>, the remaining are <strong>general</strong></p></li>
</ul>
<div class="fragment">
<p>How do we know which one to choose?</p>
</div>
</section>
<section id="bayesian-information-criterion-bic" class="slide level2">
<h2>Bayesian information criterion (BIC)</h2>
<p><strong>This is a statistical model</strong></p>
<p><span class="math display">\[f(x) = \sum_{k=1}^K \tau_k f_k(x; \theta_k) \qquad \text{where }f_k(x; \theta_k) \sim N(\boldsymbol{\mu}_k, \boldsymbol{\Sigma}_k)\]</span></p>
<div class="fragment">
<p>Use a <strong>model selection</strong> procedure for determining which best characterizes the data</p>
</div>
<div class="fragment">
<p>Specifically, use a <strong>penalized likelihood</strong> measure <span class="math display">\[\text{BIC} = 2\log \mathcal{L} - m\log n\]</span></p>
<ul>
<li><p><span class="math inline">\(\log \mathcal{L}\)</span>: log-likelihood of the considered model</p></li>
<li><p>with <span class="math inline">\(m\)</span> parameters (<em>VVV</em> has the most parameters) and <span class="math inline">\(n\)</span> observations</p></li>
</ul>
</div>
<div class="fragment">
<ul>
<li><p><strong>penalizes</strong> large models with <strong>many clusters without constraints</strong></p></li>
<li><p><strong>use BIC to choose the covariance constraints AND number of clusters</strong></p></li>
</ul>

</div>
<aside><div>
<p>The above <span class="math inline">\(\text{BIC}\)</span> is really the <span class="math inline">\(- \text{BIC}\)</span> of what you typically see, this sign flip is just for ease</p>
</div></aside></section></section>
<section>
<section id="example" class="title-slide slide level1 center">
<h1>Example</h1>

</section>
<section id="data-nba-player-statistics-per-100-possessions-2023-24-regular-season" class="slide level2">
<h2>Data: NBA player statistics per 100 possessions (2023-24 regular season)</h2>
<p>Obtained via <a href="https://github.com/rtelmore/ballr"><code>ballr</code> package</a></p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_light</span>())</span>
<span id="cb1-3"><a aria-hidden="true" tabindex="-1"></a>nba_players <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/36-SURE/2024/main/data/nba_players.csv"</span>)</span>
<span id="cb1-4"><a aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nba_players)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 33
     rk player   pos     age tm        g    gs    mp    fg   fga fgpercent   x3p
  &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt; &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;
1     1 Preciou… PF-C     24 TOT      74    18  1624   7.2  14.4     0.501   0.8
2     2 Bam Ade… C        26 MIA      71    71  2416  10.9  21       0.521   0.3
3     3 Ochai A… SG       23 TOT      78    28  1641   5.2  12.7     0.411   1.8
4     4 Santi A… PF       23 MEM      61    35  1618   7.5  17.2     0.435   3.2
5     5 Nickeil… SG       25 MIN      82    20  1921   6.1  13.8     0.439   3.4
6     6 Grayson… SG       28 PHO      75    74  2513   6.6  13.3     0.499   4  
# ℹ 21 more variables: x3pa &lt;dbl&gt;, x3ppercent &lt;dbl&gt;, x2p &lt;dbl&gt;, x2pa &lt;dbl&gt;,
#   x2ppercent &lt;dbl&gt;, ft &lt;dbl&gt;, fta &lt;dbl&gt;, ftpercent &lt;dbl&gt;, orb &lt;dbl&gt;,
#   drb &lt;dbl&gt;, trb &lt;dbl&gt;, ast &lt;dbl&gt;, stl &lt;dbl&gt;, blk &lt;dbl&gt;, tov &lt;dbl&gt;, pf &lt;dbl&gt;,
#   pts &lt;dbl&gt;, x &lt;lgl&gt;, ortg &lt;dbl&gt;, drtg &lt;dbl&gt;, link &lt;chr&gt;</code></pre>
</div>
</div>
</section>
<section id="implementation-with-mclust-package" class="slide level2">
<h2>Implementation with <a href="https://cran.r-project.org/web/packages/mclust/vignettes/mclust.html"><code>mclust</code></a> package</h2>
<p>Select the model and number of clusters</p>
<p>Use <code>Mclust()</code> function to search over 1 to 9 clusters (default) and the different covariance constraints (i.e.&nbsp;models)</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mclust)</span>
<span id="cb3-2"><a aria-hidden="true" tabindex="-1"></a><span class="co"># x3pa: 3pt attempts per 100 possessions</span></span>
<span id="cb3-3"><a aria-hidden="true" tabindex="-1"></a><span class="co"># trb: total rebounds per 100 possessions</span></span>
<span id="cb3-4"><a aria-hidden="true" tabindex="-1"></a>nba_mclust <span class="ot">&lt;-</span> nba_players <span class="sc">|&gt;</span> </span>
<span id="cb3-5"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(x3pa, trb) <span class="sc">|&gt;</span> </span>
<span id="cb3-6"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">Mclust</span>()</span>
<span id="cb3-7"><a aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(nba_mclust)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>---------------------------------------------------- 
Gaussian finite mixture model fitted by EM algorithm 
---------------------------------------------------- 

Mclust VVV (ellipsoidal, varying volume, shape, and orientation) model with 3
components: 

 log-likelihood   n df      BIC       ICL
      -2246.359 452 17 -4596.65 -4739.919

Clustering table:
  1   2   3 
 40 280 132 </code></pre>
</div>
</div>
</section>
<section id="view-clustering-summary" class="slide level2">
<h2>View clustering summary</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb5-2"><a aria-hidden="true" tabindex="-1"></a>nba_mclust <span class="sc">|&gt;</span> </span>
<span id="cb5-3"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">tidy</span>()  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 5
  component  size proportion mean.x3pa mean.trb
      &lt;int&gt; &lt;int&gt;      &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;
1         1    40     0.0822     0.101    15.5 
2         2   280     0.535      8.12      6.38
3         3   132     0.382      6.04     11.0 </code></pre>
</div>
</div>
</section>
<section id="view-clustering-assignments" class="slide level2">
<h2>View clustering assignments</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a aria-hidden="true" tabindex="-1"></a>nba_mclust <span class="sc">|&gt;</span> </span>
<span id="cb7-2"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">augment</span>() <span class="sc">|&gt;</span> </span>
<span id="cb7-3"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> x3pa, <span class="at">y =</span> trb, <span class="at">color =</span> .class, <span class="at">size =</span> .uncertainty)) <span class="sc">+</span></span>
<span id="cb7-4"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb7-5"><a aria-hidden="true" tabindex="-1"></a>  ggthemes<span class="sc">::</span><span class="fu">scale_color_colorblind</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="17-mixture_files/figure-revealjs/unnamed-chunk-9-1.png" class="quarto-figure quarto-figure-center r-stretch" width="1152"></section>
<section id="display-the-bic-for-each-model-and-number-of-clusters" class="slide level2">
<h2>Display the BIC for each model and number of clusters</h2>
<div class="columns">
<div class="column" style="text-align: left;">
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a aria-hidden="true" tabindex="-1"></a>nba_mclust <span class="sc">|&gt;</span> </span>
<span id="cb8-2"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="at">what =</span> <span class="st">"BIC"</span>, </span>
<span id="cb8-3"><a aria-hidden="true" tabindex="-1"></a>       <span class="at">legendArgs =</span> <span class="fu">list</span>(<span class="at">x =</span> <span class="st">"bottomright"</span>, <span class="at">ncol =</span> <span class="dv">4</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="17-mixture_files/figure-revealjs/nba-bic-1.png" class="quarto-figure quarto-figure-center" width="960"></p>
</figure>
</div>
</div>
</div>
</div><div class="column" style="text-align: left;">
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a aria-hidden="true" tabindex="-1"></a>nba_mclust <span class="sc">|&gt;</span> </span>
<span id="cb9-2"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="at">what =</span> <span class="st">"classification"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="17-mixture_files/figure-revealjs/nba-cluster-plot-1.png" class="quarto-figure quarto-figure-center" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="how-do-the-cluster-assignments-compare-to-the-positions" class="slide level2">
<h2>How do the cluster assignments compare to the positions?</h2>
<p>Two-way table to compare the clustering assignments with player positions</p>
<p>(What’s the way to visually compare the two labels?)</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="st">"Clusters"</span> <span class="ot">=</span> nba_mclust<span class="sc">$</span>classification, <span class="st">"Positions"</span> <span class="ot">=</span> nba_players<span class="sc">$</span>pos)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        Positions
Clusters  C C-PF PF PF-C PF-SF PG PG-SG SF SF-PF SF-SG SG SG-PG
       1 34    2  3    0     0  1     0  0     0     0  0     0
       2  3    0 33    0     1 78     4 70     1     1 88     1
       3 43    1 53    1     0  3     0 22     1     0  8     0</code></pre>
</div>
</div>
<p>Takeaway: positions tend to fall within particular clusters</p>
</section>
<section id="what-about-the-cluster-probabilities" class="slide level2">
<h2>What about the cluster probabilities?</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a aria-hidden="true" tabindex="-1"></a>nba_player_probs <span class="ot">&lt;-</span> nba_mclust<span class="sc">$</span>z</span>
<span id="cb12-2"><a aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(nba_player_probs) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cluster1"</span>, <span class="st">"cluster2"</span>, <span class="st">"cluster3"</span>)</span>
<span id="cb12-3"><a aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a aria-hidden="true" tabindex="-1"></a>nba_player_probs <span class="ot">&lt;-</span> nba_player_probs <span class="sc">|&gt;</span></span>
<span id="cb12-5"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb12-6"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">player =</span> nba_players<span class="sc">$</span>player) <span class="sc">|&gt;</span></span>
<span id="cb12-7"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>player, <span class="at">names_to =</span> <span class="st">"cluster"</span>, <span class="at">values_to =</span> <span class="st">"prob"</span>)</span>
<span id="cb12-8"><a aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a aria-hidden="true" tabindex="-1"></a>nba_player_probs <span class="sc">|&gt;</span></span>
<span id="cb12-10"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(prob)) <span class="sc">+</span></span>
<span id="cb12-11"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>() <span class="sc">+</span></span>
<span id="cb12-12"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="17-mixture_files/figure-revealjs/nba-probs-1.png" class="quarto-figure quarto-figure-center r-stretch" width="1728"></section>
<section id="which-players-have-the-highest-uncertainty" class="slide level2">
<h2>Which players have the highest uncertainty?</h2>
<div class="columns">
<div class="column" style="text-align: left;">
<p><br></p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a aria-hidden="true" tabindex="-1"></a>nba_players <span class="sc">|&gt;</span></span>
<span id="cb13-2"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cluster =</span> nba_mclust<span class="sc">$</span>classification,</span>
<span id="cb13-3"><a aria-hidden="true" tabindex="-1"></a>         <span class="at">uncertainty =</span> nba_mclust<span class="sc">$</span>uncertainty) <span class="sc">|&gt;</span> </span>
<span id="cb13-4"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">|&gt;</span></span>
<span id="cb13-5"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(uncertainty, <span class="at">n =</span> <span class="dv">5</span>) <span class="sc">|&gt;</span> </span>
<span id="cb13-6"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">player =</span> <span class="fu">fct_reorder</span>(player, uncertainty)) <span class="sc">|&gt;</span> </span>
<span id="cb13-7"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> uncertainty, <span class="at">y =</span> player)) <span class="sc">+</span></span>
<span id="cb13-8"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb13-9"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> cluster, <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">nrow =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div><div class="column" style="text-align: left;">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="17-mixture_files/figure-revealjs/unnamed-chunk-10-1.png" class="quarto-figure quarto-figure-center" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="challenges-and-resources" class="slide level2">
<h2>Challenges and resources</h2>
<ul>
<li><p>What if the data are not normal (ish) and instead are skewed?</p>
<ul>
<li><p>Apply a transformation, then perform GMM</p></li>
<li><p>GMM on principal components of data</p></li>
<li><p><a href="https://link.springer.com/article/10.1007/s11222-011-9272-x">Multivariate <span class="math inline">\(t\)</span> mixture model</a></p></li>
</ul></li>
<li><p><a href="https://www.annualreviews.org/content/journals/10.1146/annurev-statistics-033121-115326">Review paper</a>: Model-Based Clustering</p></li>
<li><p><a href="https://math.univ-cotedazur.fr/~cbouveyr/MBCbook/">Book</a>: Model-Based Clustering and Classification for Data Science</p></li>
<li><p><a href="https://sites.stat.washington.edu/raftery/Research/mbc.html">Adrian Raftery’s website</a></p></li>
</ul>
</section>
<section id="appendix-code-to-build-dataset" class="slide level2">
<h2>Appendix: code to build dataset</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb14-2"><a aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ballr)</span>
<span id="cb14-3"><a aria-hidden="true" tabindex="-1"></a>nba_players <span class="ot">&lt;-</span> <span class="fu">NBAPerGameStatisticsPer100Poss</span>(<span class="at">season =</span> <span class="dv">2024</span>)</span>
<span id="cb14-4"><a aria-hidden="true" tabindex="-1"></a>nba_players <span class="ot">&lt;-</span> nba_players <span class="sc">|&gt;</span> </span>
<span id="cb14-5"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(player) <span class="sc">|&gt;</span> </span>
<span id="cb14-6"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(g) <span class="sc">|&gt;</span> </span>
<span id="cb14-7"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb14-8"><a aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(mp <span class="sc">&gt;=</span> <span class="dv">150</span>) <span class="co"># keep players with at least 150 minutes played</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">
<p><a href="https://36-sure.github.io/2024">36-SURE.github.io/2024</a></p>
</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': true,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp("https:\/\/36-SURE\.github\.io\/2024\/");
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>